clouddriver-core/src/main/groovy/com/netflix/spinnaker/clouddriver/cache/CatsSearchProvider.groovy

```diff
@@ -128,13 +130,39 @@ class CatsSearchProvider implements SearchProvider, Runnable {
       }.flatten()
 
       if (instanceIdentifiers) {
-        cachedIdentifiersByType.set(["instances": instanceIdentifiers])
+        cachedIdentifiersByType.put("instances", instanceIdentifiers)
       }
 
       log.info("Refreshed Cached Identifiers (found ${instanceIdentifiers.size()} instances)")
     } catch (Exception e) {
       log.error("Unable to refresh cached identifiers (instances)", e)
     }
+    //Populate jobId cache only when titus is enabled
+    if (titusEnabled) {
+      try {
+        log.info("Refreshing Cached Identifiers (jobIds)")
+        def jobCache = [:]
+        def instanceIdentifiers = providers.find { provider ->
+          provider.supportsSearch('serverGroups', Collections.emptyMap())
```

Comment:
@robzienert's feedback that cloud provider specific logic should be avoided in core is valid. This specific line shows the way, in that should should be able to provide a `SearchProvider` in `clouddriver-titus` that implements the following logic.